dask/bag/tests/test_bag.py

```diff
@@ -1156,32 +1162,50 @@ def test_to_delayed():
     assert t.compute() == 21
 
 
-def test_to_delayed_optimize_graph(tmpdir):
+def test_to_delayed_optimize_graph(tmpdir, monkeypatch):
+    calls = 0
+    from dask.bag.core import reify
+
+    def count_reify(*args, **kwargs):
+        nonlocal calls
+        calls += 1
+        return reify(*args, **kwargs)
+
+    monkeypatch.setattr(dask.bag.core, "reify", count_reify)
     b = db.from_sequence([1, 2, 3, 4, 5, 6], npartitions=1)
     b2 = b.map(inc).map(inc).map(inc)
 
     [d] = b2.to_delayed()
-    text = str(dict(d.dask))
-    assert text.count("reify") == 1
```

Comment:
representations are on longer reliable like that. I migrated the test to count calls of the function using monkeypatch. Not very elegant but better overall